crates/timers/src/callback.rs

```diff
@@ -1,8 +1,25 @@
 //! Callback-style timer APIs.
 
+use js_sys::Reflect;
 use wasm_bindgen::prelude::*;
-use wasm_bindgen::JsCast;
-use web_sys::window;
+use wasm_bindgen::{throw_val, JsCast, JsValue};
+use web_sys::{Window, WorkerGlobalScope};
+
+macro_rules! global {
```

Comment:
I'd rather this be a function, it can return an enum like this:

```rust
enum WindowOrWorker {
    Window(Window),
    Worker(WorkerGlobalScope),
}
```

Then it could have a macro which would generate methods like this:

```rust
impl WorkerOrGlobal {
    fn foo(&self) {
        match self {
            Self::Window(window) => window.foo(),
            Self::Worker(worker) => worker.foo(),
        }
    }
}
```

Reply:
Oh, also, it should cache the `WindowOrWorker` inside a `thread_local!`, so that way it only needs to generate it once.

Reply:
Done, is that implementation satisfactory?